pkg/virt-launcher/virtwrap/converter/converter.go

```diff
@@ -349,30 +388,41 @@ func getOptimalBlockIOForFile(path string) (*api.BlockIO, error) {
 	}, nil
 }
 
-func SetDriverCacheMode(disk *api.Disk) error {
+func SetDriverCacheMode(disk *api.Disk, directIOChecker DirectIOChecker) error {
 	var path string
+	var err error
 	supportDirectIO := true
 	mode := v1.DriverCache(disk.Driver.Cache)
+	isBlockDev := false
 
 	if disk.Source.File != "" {
 		path = disk.Source.File
 	} else if disk.Source.Dev != "" {
 		path = disk.Source.Dev
+		isBlockDev = true
 	} else {
 		return fmt.Errorf("Unable to set a driver cache mode, disk is neither a block device nor a file")
 	}
 
 	if mode == "" || mode == v1.CacheNone {
-		supportDirectIO = checkDirectIOFlag(path)
-		if !supportDirectIO {
+		if isBlockDev {
+			supportDirectIO, err = directIOChecker.CheckBlockDevice(path)
+		} else {
+			supportDirectIO, err = directIOChecker.CheckFile(path)
+		}
+		if err != nil {
+			log.Log.Reason(err).Errorf("Direct IO check failed for %s", path)
```

Comment:
thanks. IMHO this is the place to make the decision to swallow the error.

Reply:
Hm... I thought you were insisting on propagating the error even further. Perhaps initially I misunderstood it then. I agree that its the right place.

Reply:
I cannot say that I like the current logic, but I the value of this PR keeping it. I like the current code. Unfortunately for me, I don't feel qualified to review gomock changes so I cannot lgtm it.